### DISCUSSION TITLE
Q: Why in 2.0 DateString and BlobString are not just string type?

### DISCUSSION THREAD

**Discussion by james-hu** [jsforce/jsforce#1483] [Q&A]
I am just curious, why in 2.0 branch, DateString and BlobString are defined as:
```
export type DateString = string & { __DateBrand: never };
export type BlobString = string & { __BlobBrand: never };
```

What are __DateBrand and __BlobBrand? 

Is this the expected behaviour:
```
type DateString = string & { __DateBrand: never };
const s: DateString = 'aa';
```
err:
```
Type '"aa"' is not assignable to type 'DateString'.
  Type '"aa"' is not assignable to type '{ __DateBrand: never; }'.ts(2322)
```

**Comment by stomita** [comment]
This is the [branded type](https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-branding-and-type-tagging-6cf6e516523d) to distinguish the field type in schema. It is currently only used for type inference in building query using schema info.

**Comment by kpervin** [comment]
This is causing issues with `sobject.create` calls, as no matter what I attempt, I cannot insert date data without resorting to type assertion.

**Comment by kpervin** [comment]
@stomita The `SfDate` class does not have a means to deal with these branded types, and using `SfDate.toDateLiteral(new Date())` also not not align correctly. In Typescript how are you meant to insert dates with these branded types?

Note: This issue stems from using `jsforce-gen-schema`

**Comment by camro** [comment]
Try

`sobject.create({Date__c: "2025-04-25" as DateString})`
